### PR TITLE
Added Admin Interface

### LINKED ISSUE #39: Create an IRC Admin Interface
Aoi-chan accidentally put the Imgur plugin in for Commie, so Rhe got mad at me again. He suggested adding an IRC interface / recognized commands to modify the bot while it's running. Adding / Removing plugins shouldn't be too difficult.

### PR DESCRIPTION
**Pull request by IanMitchell**
Fixes #39

### CODE REVIEW

File: src/arx7.js [hunk 1]
```diff
@@ -64,8 +64,60 @@ client.addListener('message', (from, to, text, message) => {
...
   if (to === client.nick) {
-    log(`Query from ${from}: ${text}`);
-    client.say(from, "I'm a bot! Contact Desch, Jukey, or Aoi-chan for help");
+    // Check for Admin Queries
+    if (config.admins.includes(from)) {
+      log(`Admin Query from ${from}: ${text}`);
+      let usage = `[add|remove] [plugin] [channel] [password]`;
+
+      if (text.startsWith('add ') || text.startsWith('remove ')) {
+        let args = text.split(' ');
```
Comment (kunaldes) [inline src/arx7.js]: split(/\s+/) will let you have double spaces and still get an accurate split.

File: src/arx7.js [hunk 2]
```diff
@@ -64,8 +64,60 @@ client.addListener('message', (from, to, text, message) => {
...
+
+        if (args.length != 4) {
+          log(`Invalid command count`);
```
Comment (kunaldes) [inline src/arx7.js]: command count? Drop count, maybe? idk sounds like the number of commands you gave is wrong.

File: src/arx7.js [hunk 3]
```diff
@@ -64,8 +64,60 @@ client.addListener('message', (from, to, text, message) => {
...
+          client.say(from, `Not enough commands. ${usage}`);
+        }
+
+        if (args[3] !== config.adminPassword) {
+          log(`Invalid Admin Password`);
+          client.say(from, `Invalid password. ${usage}`);
+          return;
+        }
+
+        if (config.channels[channels.indexOf(args[2])]) {
```
Comment (kunaldes) [inline src/arx7.js]: downcase args[2]

File: src/arx7.js [hunk 4]
```diff
@@ -64,8 +64,60 @@ client.addListener('message', (from, to, text, message) => {
...
+          let exists = commands.some(c => {
+            return c.constructor.name.toLowerCase() === args[1];
```
Comment (kunaldes) [inline src/arx7.js]: Actually just downcase everything but the password.

File: src/arx7.js [hunk 5]
```diff
@@ -64,8 +64,60 @@ client.addListener('message', (from, to, text, message) => {
...
+          });
+
+          if (exists) {
+            if (args[0] === 'add') {
+              log(`ENABLE command for ${args[1]} in ${args[2]}`);
+              config.channels[channels.indexOf(args[2])].plugins.push(args[1]);
```
Comment (kunaldes) [inline src/arx7.js]: This doesn't check if it's already enabled? won't that cause it to reply multiple times to every command?

**Comment by IanMitchell** [comment]
Updated - code is a little messy, will be cleaning it up as part of a refactor

**Comment by IanMitchell** [comment]
Eh, I'll just fix it before merging it.

**Pull request closed without merging**